Sources/Sentry/SentryDebugImageProvider.m

```diff
@@ -36,9 +39,24 @@ - (instancetype)initWithBinaryImageProvider:(id<SentryCrashBinaryImageProvider>)
     if (self = [super init]) {
         self.binaryImageProvider = binaryImageProvider;
     }
+
+    _dyld_register_func_for_add_image(clearCachedDebugImages);
+    _dyld_register_func_for_remove_image(clearCachedDebugImages);
+
```

Comment:
In some cases your callback function may be called on a different thread (namely the thread that is loading the library), see here: https://twitter.com/gparker/status/1266572192126386176?s=20&t=jXOtWHAPuZ8oUrlUj788jQ

In this scenario the callback functions are accessing `cachedDebugImages` in a non-thread safe way, so you would need to use a lock to guard all reads/writes to `cachedDebugImages`.